lib/matplotlib/dates.py

```diff
@@ -604,6 +605,17 @@ def _wrap_in_tex(text):
     return ret_text
 
 
+def _get_tzinfo(tz):
+    """Generate tzinfo from a string or return tzinfo."""
+    if tz is None:
+        return _get_rc_timezone()
+    if isinstance(tz, str):
```

Comment:
If we do this, maybe a `try/except` here that says the string has to be accepted by `dateutil`?  Currently the user will get a dateutil error, but often upstream error messages are mystifying to people.  

I think re-implimenting this is fine, we just need to be explicit about what strings are accepted so people know what to google to see the list.  Probably the docs at the top of the module should be updated to point to this as well?

Reply:
Turns out that `dateutil.tz.gettz` returns `None` if not working out (just tested as I had no idea). But makes sense to raise an exception in that case.

Reply:
Lets try that - on the one hand strings have been broken for ever, on the other they would be convenient for users.